### PR TITLE
K8s: Add k8s codegen PR check

### PR DESCRIPTION
**Pull request by toddtreece**
**What is this feature?**

Follow up to #91867 to add a k8s codegen test and fix codegen for `pkg/apimachinery/apis/common/v0alpha1`

**Why do we need this feature?**

`go mod download` was required to ensure that `code-generator` had the `meta/v1` dependencies required for `kube-openapi` codegen. `pkg/apimachinery/apis/common/v0alpha1` is the only API that currently includes the `meta/v1` package.

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by ryantxu — APPROVED** [review]
LGTM -- but I am always nervous to touch the alerting side

File: pkg/generated/applyconfiguration/alerting_notifications/v0alpha1/integration.go
Comment (toddtreece) [inline pkg/generated/applyconfiguration/alerting_notifications/v0alpha1/integration.go:16]: @ryantxu this is an interesting change in the generated code

**Comment by toddtreece** [comment]
@yuri-tceretian have you seen this error before? https://github.com/grafana/grafana/actions/runs/10389637679/job/28768118009?pr=91903#step:4:232

**Comment by toddtreece** [comment]
> LGTM -- but I am always nervous to touch the alerting side

@ryantxu i'll wait until someone from @grafana/alerting-backend approves

**Comment by yuri-tceretian** [comment]
> @yuri-tceretian have you seen this error before?

yes, it started to pop up a while ago. I could not figure out what exactly it complains about.

**Comment by toddtreece** [comment]
> I could not figure out what exactly it complains about.

@yuri-tceretian i'll take a look and see if i can figure it out. since this is not an error that was introduced in this PR, could you double check the alerting changes and approve if everything looks ok?

File: pkg/generated/applyconfiguration/alerting_notifications/v0alpha1/integration.go
Comment (yuri-tceretian) [reply pkg/generated/applyconfiguration/alerting_notifications/v0alpha1/integration.go:16]: Perhaps that happened because I forgot to regenerate code in https://github.com/grafana/grafana/pull/91430

**Review by yuri-tceretian — APPROVED** [review]
LGTM

File: pkg/generated/applyconfiguration/alerting_notifications/v0alpha1/integration.go
Comment (toddtreece) [reply pkg/generated/applyconfiguration/alerting_notifications/v0alpha1/integration.go:16]: ah ok, that makes sense

**Comment by diegommm** [comment]
Hi! Greetings from the Release Guild. This is a friendly reminder that the Steady [11.2.x Release Branch](https://github.com/grafana/grafana/tree/v11.2.x) was cut this week, and that it includes the changes committed up to https://github.com/grafana/grafana/commit/b4126d3bce3627e9460e2db0e870c8d984be8e84. This PR was merged after it, so we wanted to let you know that you can add the `backport v11.2.x` label to to backport it and make it part of the next 11.2.x Patch Release, just in case you may have missed the news.

Thank you!

**Pull request merged**
